File: NET_8/FoodCall.Domain/Entities/OrderItem.cs
namespace FoodCall.Domain.Entities;

public class OrderItem
{
    public Guid Id { get; private set; }
    public Guid ProductId { get; private set; }
    public string ProductName { get; private set; }
    public decimal UnitPrice { get; private set; }
    public int Quantity { get; private set; }
    
    public decimal SubTotal => UnitPrice * Quantity;

    public OrderItem(Guid productId, string productName, decimal unitPrice, int quantity)
    {
        if (quantity <= 0)
            throw new ArgumentException("A quantidade deve ser maior que zero.");

        if (unitPrice < 0)
            throw new ArgumentException("O preço unitário não pode ser negativo.");

        Id = Guid.NewGuid();
        ProductId = productId;
        ProductName = productName;
        UnitPrice = unitPrice;
        Quantity = quantity;
    }
}

File: Java/src/main/java/com/foodcall/domain/entities/OrderItem.java
package com.foodcall.domain.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Table(name = "order_items")
@Getter
@NoArgsConstructor
public class OrderItem {
    
    @Id
    private UUID id;
    
    @Column(nullable = false)
    private UUID productId;
    
    @Column(nullable = false, length = 200)
    private String productName;
    
    @Column(nullable = false)
    private double unitPrice;
    
    @Column(nullable = false)
    private int quantity;
    
    public OrderItem(UUID productId, String productName, double unitPrice, int quantity) {
        if (quantity <= 0) {
            throw new IllegalArgumentException("A quantidade deve ser maior que zero.");
        }
        if (unitPrice < 0) {
            throw new IllegalArgumentException("O preço unitário não pode ser negativo.");
        }
        
        this.id = UUID.randomUUID();
        this.productId = productId;
        this.productName = productName;
        this.unitPrice = unitPrice;
        this.quantity = quantity;
    }
    
    public double getSubTotal() {
        return unitPrice * quantity;
    }
}